### PR TITLE
ocm: add thanos repo

### PR DESCRIPTION
**Pull request by songleo**
https://github.com/open-cluster-management/backlog/issues/10609

### CODE REVIEW

**Review by petr-muller — APPROVED** [review]

File: ci-operator/config/open-cluster-management/thanos/open-cluster-management-thanos-release-2.3.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      memory: 1Gi
+tag_specification:
+  name: "4.7"
+  namespace: ocp
+test_binary_build_commands: "true"
+tests:
+- as: e2e-aws
+  steps:
+    cluster_profile: aws
+    workflow: openshift-e2e-aws
```
Comment (petr-muller) [inline ci-operator/config/open-cluster-management/thanos/open-cluster-management-thanos-release-2.3.yaml:41]: Actually, what's the point of having a presubmit that 1) installs a vanilla OCP and 2) runs shared OCP tests againts the installed cluster. None of your code is tested by this, why have it?

/hold

Comment (songleo) [reply ci-operator/config/open-cluster-management/thanos/open-cluster-management-thanos-release-2.3.yaml:41]: Thank you for pointing it out. yes, I should remove it.

Comment (songleo) [reply ci-operator/config/open-cluster-management/thanos/open-cluster-management-thanos-release-2.3.yaml]: hi @petr-muller , I have updated PR, pls help to review it. thanks.

**Review by petr-muller — APPROVED** [review]

**Pull request merged**